src/components/CurrencyTable/CurrencyTable.js

```diff
@@ -1,70 +1,79 @@
-
-    const span = document.createElement("span");
-    span.innerHTML = "✕";
-    span.classList.add("cross");
-    span.classList.add("icon");
-
-    parent.appendChild(span);
-
-    span.addEventListener("click", () => {
-      span.style.display = "none";
-      child.style.display = "inline-block";
-      input.disabled = true;
-    });
-
-    input.disabled = false;
+  const handleClickEdit = (rate, ccy, operation) => {
+    setActiveInput(true);
+    setActiveCCY(ccy);
+    setActiveType(operation);
+    setActiveNewValue(rate);
   };
 
-  const changeValue = (e) => {
-    const currencyValue = e.target.value;
-    setNewValue(currencyValue);
+  const handleSave = () => {
+    setActiveInput(false);
+    dispatch(updateValue({ ccy: activeCCY, operation: activeType, rate: activeValue }));
   };
 
-  const resetStorageValues = () => {
-    localStorage.clear();
+  const handleClose = () => {
+    setActiveInput(false);
+    setActiveCCY("");
+    setActiveType("");
+    setActiveNewValue("");
   };
 
-  const table = document.getElementsByClassName("table");
+  const changeValue = (event) => {
+    const currencyValue = event.target.value;
+    setActiveNewValue(currencyValue);
+  };
 
-  if (localStorage.getItem("count") === "5") {
-    if (table.length > 0) {
-      table[0].style.display = "none";
-    }
+  const renderButtons = () => {
     return (
-      <div className="error">
-        <p className="error-message">The localStorage is full</p>
-        <button className="reset" onClick={resetStorageValues}>
-          Reset localStorage
-        </button>
-      </div>
+      <>
+        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" className="check-mark" onClick={handleSave}>
+          <path d="M470.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L192 338.7 425.4 105.4c12.5-12.5 32.8-12.5 45.3 0z" />
+        </svg>
+        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512" className="cross-icon" onClick={handleClose}>
+          <path d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z" />
+        </svg>
+      </>
     );
-  }
+  };
+
+  const renderOptions = (rate, operation, ccy) => {
+    if (activeInput === true && activeCCY === ccy && activeType === operation) {
+      return (
+        <>
+          <input className="input" value={activeValue} onChange={changeValue} />
+          {renderButtons()}
+        </>
+      );
+    } else {
+      return (
+        <>
+          <input className="input" value={rate} disabled />
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" className="edit-icon" onClick={() => handleClickEdit(rate, ccy, operation)}>
```

Comment:
I believe you can import svg as component https://blog.logrocket.com/how-to-use-svgs-react/